tests/unit/test_charm_2.py

```diff
@@ -463,42 +552,81 @@ def test_theme_reconciliation(self):
         assert: themes installed in WordPress should update according to the themes config
         """
         self._setup_replica_consensus()
-        self.harness.update_config({
+        db_config = {
             "db_host": "config_db_host",
             "db_name": "config_db_name",
             "db_user": "config_db_user",
             "db_password": "config_db_password",
-        })
+        }
+        self.patch.allow_database(db_config)
+        self.harness.update_config(db_config)
 
         self.assertEqual(
-            self.installed_themes,
+            self.patch.installed_themes(),
             set(self.harness.charm._WORDPRESS_DEFAULT_THEMES),
             "installed themes should match the default installed themes "
             "with the default themes config"
         )
 
-        # Currently, Ops test framework has a bug preventing starting an already
```

Comment:
Has this bug been resolved?

Reply:
No, I just add some code in charm to check activeness before start or stop a service, or there will be too much patches like this in unit tests.